thunder/benchmarks/benchmark_litgpt.py

```diff
@@ -101,6 +184,10 @@ def __init__(
         warmup_iters: int = 25,
         dump_thunder_traces: bool = False,
         dump_memory_snapshot: bool = False,
+        use_torchao_fp8_linear: bool = False,
+        use_torchao_fp8_allgather: bool = False,
+        # FIXME(crcrpar): loss goes nan after one param update.
```

Comment:
cc @weifengpy on the nan issue

Reply:
quick check before deep dive: if we turn off torch.compile, will the loss go nan? is loss nan because of `nan` in `model.parameters()` before 1st `optimizer.step` or some `.grad` becomes `nan` before 1st `optimizer.step`?

Reply:
I am excited to for the benchmarking!

Reply:
> if we turn off torch.compile, will the loss go nan?

I didn't check it. But it seems to lead to dtype mismatch error:

If I keep the model in float32, things look okay.
Would having the model in bfloat16 happen to have some effects on precompute?
https://github.com/Lightning-AI/lightning-thunder/blob/16e79766bd29ff9e22637f82f3f4ab0ac0037130/thunder/benchmarks/benchmark_litgpt.py#L251-L256

A smaller reproducible script is here: https://gist.github.com/crcrpar/23b5b33418e5e11d668280678664c03a

Reply:
thanks for the minimized repro. I can give it try

> But it seems to lead to dtype mismatch error

just to clarify, is it runtime error or loss nan? At first I thought I should debug loss nan without runtime error

Reply:
I'm seeing
```
# --- eager ---
# [rank0]:[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torchao/float8/float8_linear.py", line 363, in forward
# [rank0]:[rank0]:     output = manual_float8_matmul.apply(input_fp8, weight_fp8.t())
# [rank0]:[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torch/autograd/function.py", line 575, in apply
# [rank0]:[rank0]:     return super().apply(*args, **kwargs)  # type: ignore[misc]
# [rank0]:[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torchao/float8/float8_linear.py", line 60, in forward
# [rank0]:[rank0]:     res_bits = torch.mm(input_fp8_reshaped, weight_fp8_t)
# [rank0]:[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torchao/float8/float8_tensor.py", line 359, in __torch_dispatch__
# [rank0]:[rank0]:     return FLOAT8_OPS_TABLE[func](func, args, kwargs)
# [rank0]:[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torchao/float8/float8_ops.py", line 181, in float8_mm
# [rank0]:[rank0]:     tensor_out = addmm_float8_unwrapped(
# [rank0]:[rank0]:   File "/usr/local/lib/python3.10/dist-packages/torchao/float8/float8_python_api.py", line 54, in addmm_float8_unwrapped
# [rank0]:[rank0]:     output = torch._scaled_mm(
# [rank0]:[rank0]: RuntimeError: Both scale_a and scale_b must be float (fp32) tensors.
```
(same as in the gist)